crates/edr_solidity/src/contracts_identifier/radix_tree.rs

```diff
@@ -0,0 +1,416 @@
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct RadixNode {
+    content: Vec<u8>,
+    is_present: bool,
+    bytes_matched_before: usize,
+    child_nodes: HashMap<u8, RadixNode>,
+}
+
+impl RadixNode {
+    fn new(content: Vec<u8>, is_present: bool, bytes_matched_before: usize) -> RadixNode {
+        RadixNode {
+            content,
+            is_present,
+            bytes_matched_before,
+            child_nodes: HashMap::new(),
+        }
+    }
+
+    pub fn get_content(&self) -> &Vec<u8> {
+        &self.content
+    }
+
+    fn add_word(&mut self, word: Vec<u8>) {
```

Comment:
Should we not signal that we (un)succesfully added the word?

Reply:
I don't think there's a concept of "failure" here. The only thing we could signal is if the word was already there or not. But since (currently) no callee needs that, I wouldn't add it for now.

Reply:
If the word is empty, we don't do anything. That seems like it's breaking the user's expectation to me:

1. I call `foo.add_word(word)`
2. The function returns without panicking. As the function signature doesn't communicate that there is a possibility my word won't actually be added to the `radix_tree` I assume it has been added. In fact, it has not.
3. If I call `get_max_match`, I think the result will be a `false` boolean as `entire_word_matched == true` but `entire_content_matched == false`

I would consider this discrepancy between expectation and reality to be a failure mode

Reply:
In that case I would prefer to just panic when an empty word is passed. 

If we return a `bool`, then we should also analyze what happens when a word that already exists is added; for example, `HashSet` returns `false` in that case. This means that there would be three cases here: the word was not added because it was empty, the word was not added because it already existed, and the word was added. Since I don't think it's worth spending time on that analysis, I would just panic for empty words.

Reply:
You can return a `SomeError` and `Result<(), SomeError>` from that function. At leat that'll allow people to recover from the error